src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -230,6 +231,23 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
     }
 
+    void MainPage::_Navigate(const Editor::ProfileViewModel& profile)
+    {
+        auto state{ winrt::make_self<ProfilePageNavigationState>(profile, _settingsClone.GlobalSettings().ColorSchemes(), *this) };
+
+        // You cannot delete the Windows Powershell and CMD profiles
+        const auto myGuid{ profile.Guid() };
+        const winrt::guid pwshGuid{ ::Microsoft::Console::Utils::GuidFromString(L"{61c54bbd-c2c6-5271-96e7-009a87ff44bf}") };
+        const winrt::guid cmdGuid{ ::Microsoft::Console::Utils::GuidFromString(L"{0caa0dad-35be-5f56-a8ff-afceeeaa6101}") };
+        if (myGuid != pwshGuid && myGuid != cmdGuid)
+        {
+            auto pfnDeleteProfile{ std::bind(&MainPage::_DeleteProfile, this, profile) };
+            state->SetDeleteProfileCallback(pfnDeleteProfile);

```

Comment:
Okay so this whole thing is _weird_. I get it, but I think it needs some comments. Lemme make sure I followed this right:
* The MainPage has access to the whole settings model, but the profile page only has access to the Profile. So if the profile page wants to be able to delete a profile, it needs to ask the MainPage to do it.
* Not all profiles are deletable (namely, `cmd` and `powershell`).
* We're passing in a callback to the profile page if the profile is deletable.
* When the delete button is pressed, we'll call that callback, and MainPage will delete the profile.

<hr>

1. Is there any reason we didn't just go with a normal event for this? Something like `Profiles::DeleteRequested`, that `MainPage` listens for? 
2. Should the availability of the delete button be in the Profile ViewModel? Seems weird to make the MainPage responsible for it here. Esp. considering the future where dynamic profiles (and defaults) might need to be "hidden", not "deleted"

Reply:
Your explanation is spot on. I'll add a comment documenting the whole thing.

> 1. Is there any reason we didn't just go with a normal event for this? Something like `Profiles::DeleteRequested`, that `MainPage` listens for? 

No particular reason. I'm not familiar with the benefits/tradeoffs of each approach though so I'm open to changing it to that if there's a good reason.

> 2. Should the availability of the delete button be in the Profile ViewModel? Seems weird to make the MainPage responsible for it here. Esp. considering the future where dynamic profiles (and defaults) might need to be "hidden", not "deleted"

Hmm... I guess the design would be something like this...
- Profiles.cpp holds a list of in-box profile guids and dynamic profile guids
- bind DeleteButton.IsEnabled to a guid-check
- present a text block with why it's disabled based on the guid check

I'm sold! Changing it to this design.